### ISSUE TITLE
Adding support for new boards without compiling TinyGo from scratch

### ISSUE THREAD

**Issue by oxplot** [tinygo-org/tinygo#3812]
As I understand it, the only way to add support for new boards is to add certain board specific files and compile TinyGo. Is there plan to remove this requirement? Having to compile the entire thing is not in itself a big deal, but having to maintain a fork of the upstream is the time consuming part.

EDIT: In addition, any user of the board will have to use the fork and many would rather not.

**Comment by oxplot** [comment]
Never mind — I missed that `-target` can take JSON spec file.

**Comment by oxplot** [comment]
Spoke too soon. I can specify the JSON spec file but the extra files (in case of RP2040, the second stage bootloader) is only loaded from Tiny Go root path.

**Comment by aykevl** [comment]
Yeah this is something we've wanted to add for a long time, but there is no single obvious way this can be implemented.
